src/test/java/org/icgc/argo/program_service/services/EgoServiceIT.java

```diff
@@ -142,6 +143,34 @@ void setUp() {
     assertThat(adminPermissions).allMatch(permission -> permission.getAccessLevel().equals("WRITE"));
   }
 
+  @Test
+  public void updateUser(){
+    val user = egoService.getUser("Test_update_user_PS@dummy.com");
+    assertThat(user.isPresent()).isTrue();
+
+    // add user to COLLABORATOR group
+    assertThat(egoService.joinProgram(UPDATE_USER_TEST_EMAIL, programEntity, UserRole.COLLABORATOR)).isTrue();
+
+    val groupBefore = egoService.getGroupsFromUser(user.get().getId());
+    assertThat(groupBefore.isEmpty()).isFalse();
+    assertThat(groupBefore.size()).isEqualTo(1);
+    groupBefore.forEach(group -> assertThat(group.getName()).isEqualTo("PROGRAM-TestShortName-COLLABORATOR"));
+
+    // expected group is ADMIN group
+    egoService.updateUserRole(user.get().getId(), programEntity.getId(), UserRole.ADMIN);
+    val adminGroupName = "PROGRAM-TestShortName-ADMIN";
+    val adminGroupId = egoService.getGroup(adminGroupName).get().getId();
+
+    // verify if user role is updated to ADMIN
+    val groupAfter = egoService.getGroupsFromUser(user.get().getId());
+    assertThat(groupAfter.isEmpty()).isFalse();
+    assertThat(groupAfter.size()).isEqualTo(1);
+    groupAfter.forEach(group -> assertThat(group.getId()).isEqualTo(adminGroupId));
+
+    // remove test user from admin group
+    assertThat(egoService.leaveProgram(user.get().getId(), programEntity.getId())).isTrue();
```

Comment:
unfortunately, the `leaveProgram` method will return `true` even if a `HttpServerErrorException` or `HttpClientErrorException` is thrown. An additional check should be made to ensure the user doesnt belong to the group